File: debop4k-core/src/test/java/debop4k/core/asyncs/BackgroundWorkerTest.java
/*
 * Copyright (c) 2016. Sunghyouk Bae <sunghyouk.bae@gmail.com>
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 */

package debop4k.core.asyncs;

import debop4k.core.AbstractCoreTest;
import lombok.extern.slf4j.Slf4j;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.test.context.junit4.SpringRunner;

import javax.inject.Inject;

import static org.assertj.core.api.Assertions.assertThat;

@Slf4j
@RunWith(SpringRunner.class)
@SpringBootTest(classes = {BackgroundWorkerConfiguration.class})
public class BackgroundWorkerTest extends AbstractCoreTest {

  @Inject DummyBackgroundWorker worker;

  @Test
  public void testDummyBackgroundWorker() throws Exception {
    assertThat(worker).isNotNull();
    Thread.sleep(100L);
    assertThat(worker.isRunning()).isTrue();
    Thread.sleep(100L);
  }
}

File: debop4k-core-java6/src/test/kotlin/debop4k/core/asyncs/BackgroundWorkerTest.kt
/*
 * Copyright (c) 2016. Sunghyouk Bae <sunghyouk.bae@gmail.com>
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package debop4k.core.asyncs

import debop4k.core.uninitialized
import org.assertj.core.api.Assertions.assertThat
import org.junit.Test
import org.junit.runner.RunWith
import org.slf4j.LoggerFactory
import org.springframework.boot.test.SpringApplicationConfiguration
import org.springframework.test.context.junit4.SpringJUnit4ClassRunner
import javax.inject.Inject

/**
 * BackgroundWorkerTest
 * @author debop sunghyouk.bae@gmail.com
 */
@RunWith(SpringJUnit4ClassRunner::class)
@SpringApplicationConfiguration(classes = arrayOf(BackgroundWorkerConfiguration::class))
class BackgroundWorkerTest {

  private val log = LoggerFactory.getLogger(javaClass)

  @Inject val worker: DummyBackgroundWorker = uninitialized()

  @Test
  fun injectDummyBackgroundWorker() {
    assertThat(worker).isNotNull()
    Thread.sleep(500)
    assertThat(worker.isRunning).isTrue()
    Thread.sleep(500)
  }
}